Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Isotropic_remeshing/remesh_impl.h

```diff
@@ -281,22 +341,34 @@ namespace internal {
       : mesh_(pmesh)
       , vpmap_(vpmap)
       , own_tree_(own_tree)
+      , has_border_(false)
       , input_triangles_()
       , input_patch_ids_()
-      , halfedge_status_map_()
+#ifndef SM_HALFEDGE_STATUS_PMAP
+      , halfedge_status_pmap_(0)
+#endif
       , protect_constraints_(protect_constraints)
       , patch_ids_map_(fpmap)
       , ecmap_(ecmap)
       , vcmap_(vcmap)
       , fimap_(fimap)
     {
       CGAL_assertion(CGAL::is_triangle_mesh(mesh_));
+#ifdef SM_HALFEDGE_STATUS_PMAP
+  halfedge_status_pmap_ = mesh_. template add_property_map<halfedge_descriptor,Halfedge_status>("h:status", MESH).first;
+#endif
     }
 
     ~Incremental_remesher()
     {
-      if (own_tree_)
-        delete tree_ptr_;
+#ifdef SM_HALFEDGE_STATUS_PMAP
+      mesh_.remove_property_map(halfedge_status_pmap_);
+#endif
+      if (own_tree_){
+        for(int i=0; i < trees.size();++i){
+          delete trees[i];
+        }
+      }
```

Comment:
The role of `own_tree_` is still very obscure. If it is set, there is probably a memory leak. @janetournois, I think I found one place in the demo where that Boolean is set to true.